spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/sql/TestCreateTableAsSelect.java

```diff
@@ -100,6 +101,15 @@ public void testPartitionedCTAS() {
         "Should have rows matching the source table",
         sql("SELECT * FROM %s ORDER BY id", sourceName),
         sql("SELECT * FROM %s ORDER BY id", tableName));
+
+    // Write distribution mode is not respected before Spark 3.5.0
+    Assert.assertThrows(
```

Comment:
this should use `Assertions.assertThatThrownBy(() -> ...).isInstanceOf(SparkException.class).hasMessage(...)`

Reply:
also it's probably better to just add a new test method: `ctasWriteDistributionModeNotRespected()`